adagrams/game.py

```diff
@@ -1,11 +1,91 @@
+import random
+import copy
+from collections import Counter
+
+LETTER_POOL = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+}
+
+
+score_table = {
+    1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+    2: ["D", "G"],
+    3: ["B", "C", "M", "P"],
+    4: ["F", "H", "V", "W", "Y"],
+    5: ["K"],
+    8: ["J", "X"],
+    10: ["Q", "Z"]
+}
+
+
 def draw_letters():
-    pass
+    random_list = []
+    for letter, frequency in LETTER_POOL.items():
+        random_list += list(letter * frequency)
+    return random.sample(random_list, 10)
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_bank_copy = copy.deepcopy(letter_bank)
+    for letter in word.upper():
+        if letter in letter_bank_copy:
+            letter_bank_copy.remove(letter)
```

Comment:
Binary search is used to find a value in ordered data. We _could_ use it for each list of letters, assuming they are stored in order, to improve the speed of looking up a letter in the current score bin, but we still need to iterate over all the scores.

Instead, we could store the score data as a dictionary of letter->score. Just because the score table was written out as score->letter list doesn't mean that's the way we are required to represent it in the program.

So instead of 1->[A, E, I, ...], 2->[D, G], etc, we could flip the relationships around to have A->1, E->1, I->1, D->2, G->2, etc. Then to get the score of a letter, we just look it up using the letter as the key.